pxtblocks/blocklycompiler.ts

```diff
@@ -212,23 +212,31 @@ namespace pxt.blocks {
                 // HACK: The real type is stored as the second check
                 return ground(b.outputConnection.check_[1])
             }
-            // The only block that hits this case should be lists_create_with, so we
-            // can safely infer the type from the first input that has a return type
+            // lists_create_with and argument_reporter_array both hit this.
+            // For lists_create_with, we can safely infer the type from the
+            // first input that has a return type.
+            // For argument_reporter_array just return any[] for now
             let tp: Point;
-            if (b.inputList && b.inputList.length) {
-                for (const input of b.inputList) {
-                    if (input.connection && input.connection.targetBlock()) {
-                        let t = find(returnType(e, input.connection.targetBlock()))
-                        if (t) {
-                            if (t.parentType) {
-                                return t.parentType;
+            if (b.type == "lists_create_with") {
+                if (b.inputList && b.inputList.length) {
+                    for (const input of b.inputList) {
+                        if (input.connection && input.connection.targetBlock()) {
+                            let t = find(returnType(e, input.connection.targetBlock()))
+                            if (t) {
+                                if (t.parentType) {
+                                    return t.parentType;
+                                }
+                                tp = ground(t.type + "[]");
+                                genericLink(tp, t);
+                                break;
                             }
-                            tp = ground(t.type + "[]");
-                            genericLink(tp, t);
-                            break;
                         }
                     }
                 }
+            } else if (b.type =="argumet_reporter_array") {
```

Comment:
```suggestion
            } else if (b.type == "argument_reporter_array") {
```

is the typo here intentional / an existing typo? `argumet` vs `argument`

Reply:
>> nope. thanks joey!